settings/src/components/email-address.js

```diff
@@ -0,0 +1,94 @@
+
+/**
+ * WordPress dependencies
+ */
+import { Button, TextControl, Notice, Spinner } from '@wordpress/components';
+import { useState, useEffect } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as coreDataStore, useEntityRecord } from '@wordpress/core-data';
+
+/**
+ * Render the Email setting.
+ */
+export default function EmailAddress( { userId } ) {
+	const { record, edit, save, editedRecord, hasEdits } = useEntityRecord( 'root', 'user', userId );
+
+	// useEntityRecord() doesn't expose isSaving. Huh.
+	const isSaving = useSelect( ( select ) => select( coreDataStore ).isSavingEntityRecord( 'root', 'user', userId ) );
+
+	const [ emailError, setEmailError ] = useState( '' );
+	const [ justChangedEmail, setJustChangedEmail ] = useState( false );
+
+	const handleEmailChange = ( email ) => edit( { email } );
+
+	const handleSave = async () => {
+		try {
+			await save();
+
+			setJustChangedEmail( true );
+		} catch( error ) {
+			// TODO: Change these texts
+			// error.code: rest_user_invalid_email (can't use that one), rest_invalid_param (invalid email format)
+			// TODO: The red paragraph inserted inline feels a bit hacky.
+			setEmailError( error.message );
+		}
+	};
+
+	const handleDiscard = async () => {
+		try {
+			await edit( { pending_email: '' } );
+			await save();
+		} catch( error ) {
+			alert( error.message );
+		}
+	};
+
+	if ( ! record ) {
+		return <Spinner />
+	}
+
+	return (
+		<>
+			{ record.pending_email && ! justChangedEmail && <Notice status="warning" className="actions-on-right" isDismissible={ false } actions={ [ { label: "Cancel change", onClick: handleDiscard } ] }>
```

Comment:
Agreed!

I was also wondering if there was a need here to use a centralised noticeStore instead or something.. I've seen examples of the Editor interface having a single notice spot, rather than it being handled by individual components, but I guess since the notices are shown within the components, it makes sense to keep it here for now.